src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,33 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    public static final int SET_VALUE = 10;
+    public static final int MAX_FOR_FIGURE = 15;
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    private Figure[] getArrayFigures() {
+        Figure circle = new Circle(colorSupplier.getRandomColor(),random.nextInt(MAX_FOR_FIGURE));
+        Figure trapezoid = new IsoscelesTrapezoid(colorSupplier.getRandomColor(),
+                random.nextInt(MAX_FOR_FIGURE),random.nextInt(MAX_FOR_FIGURE),
+                random.nextInt(MAX_FOR_FIGURE));
+        Figure rectangle = new Rectangle(colorSupplier.getRandomColor(),
+                random.nextInt(MAX_FOR_FIGURE),random.nextInt(MAX_FOR_FIGURE));
+        Figure triangle = new RightTriangle(colorSupplier.getRandomColor(),
+                random.nextInt(MAX_FOR_FIGURE),random.nextInt(MAX_FOR_FIGURE));
+        Figure square = new Square(colorSupplier.getRandomColor(),random.nextInt(MAX_FOR_FIGURE));
+        Figure[] figures = new Figure[]{circle,trapezoid,rectangle,triangle,square};
+        return figures;
+    }
+
+    public Figure getRandomFigure() {
+        int index = random.nextInt(getArrayFigures().length);
+        return getArrayFigures()[index];
+    }
```

Comment:
With such approach you're creating 5 figures each time method is called while you need only one of them.
Please reconsider logic so you firstly determine type of random figure and only then create this particular figure without creation of all other ones